pkgs/development/libraries/arrow-cpp/default.nix

```diff
@@ -228,14 +237,35 @@ stdenv.mkDerivation rec {
     in
     lib.optionalString doInstallCheck "-${builtins.concatStringsSep ":" filteredTests}";
   installCheckInputs = [ perl which sqlite ] ++ lib.optional enableS3 minio;
+
+  postBuild = lib.optionalString (doInstallCheck && enableCuda) ''
+    for file in $(find . -name '*-test' -o -name '*-tests' -type f -executable); do
+      addOpenGLRunpath "$file"
+    done
+  '';
+
   installCheckPhase =
     let
       excludedTests = lib.optionals stdenv.isDarwin [
         # Some plasma tests need to be patched to use a shorter AF_UNIX socket
         # path on Darwin. See https://github.com/NixOS/nix/pull/1085
         "plasma-external-store-tests"
         "plasma-client-tests"
-      ] ++ [ "arrow-gcsfs-test" ];
+      ] ++ [
+        "arrow-gcsfs-test"
+      ] ++ lib.optionals enableCuda [
+        # we can run this but it requires access to /run/opengl-driver and some
+        # other things under /dev, see https://github.com/NixOS/nix/issues/4489
+        "arrow-cuda-test"
+        # these transitively link in libcuda, so we can't run them if arrow
+        # is being built with cuda support
+        "arrow-flight-internals-test"
+        "arrow-flight-sql-test"
+        "arrow-flight-test"
+        "plasma-client-tests"
+        "plasma-external-store-tests"
+        "plasma-serialization-tests"
```

Comment:
If these tests dynamically link against libcuda.so but don't actually invoke `cuInit` you may be able to get away with using the "stub" version of libcuda.so provided in CUDA...

But I don't think that ought to be a blocker for merging this.

Reply:
Neat, I will look into that.

Reply:
You can see how pytorch uses this trick [here](https://cs.github.com/nathyong/nixpkgs/blob/63684d50be5f52764c1e866db52cb623df346f76/pkgs/development/python-modules/pytorch/default.nix#L20), albeit with the outdated `cudatoolkit` package.

Reply:
Awesome, I'll copy that over and adapt it.